src/board/ui.ts

```diff
@@ -397,6 +404,65 @@ export class AccelerometerUI {
   dispose() {}
 }
 
+export class MicrophoneUI {
+  public soundLevel: RangeSensor = new RangeSensor(
+    "soundLevel",
+    0,
+    255,
+    0,
+    undefined
+  );
+  private lowThreshold: number;
+  private highThreshold: number;
+  private thresholdState: "lowThreshold" | "highThreshold" | "none";
```

Comment:
Unsure whether using `thresholdState` is the right approach.

Reply:
Rejigged to have access to previous and current values.